Automattic/jetpack
# Connection: Add site creation date argument

As part of our Jetpack Improved Onboarding work, we'd like to send an additional argument with the rest of the connection URL arguments. That argument would contain the site creation date, which is basically the earliest of either: 

* The earliest admin user registration date.
* The earliest post date.

Companion diff: D25012-code.

#### Changes proposed in this Pull Request:

* Introduce a `Jetpack::get_assumed_site_creation_date()` method.
* Add a new `site_created` parameter to the connection URLs.
* Add an exception for multisite subsites, where we use the multisite creation date instead.

#### Testing instructions:

* Follow the testing instructions in D25012-code.
* Checkout this branch.
* Verify tests pass by running either:

```
yarn docker:phpunit --filter test_get_assumed_site_creation_date
```

or:

```
phpunit --filter test_get_assumed_site_creation_date
```

depending on the environment you're developing Jetpack in.

#### Proposed changelog entry for your changes:

* No changelog entry needed.
